### PR TITLE
Add support for custom ini directives

### PR DESCRIPTION
**Pull request by gglnx**
The PHP cli allows you with `-d` to add custom [ini directives](http://php.net/manual/en/ini.list.php). Great for a custom error log path or others settings without losing the default `php.ini` system settings. This pull requests add a new option called <s>`settings`</s>`directives` for that.

### CODE REVIEW

File: tasks/php.js [hunk 1]
```diff
@@ -72,6 +73,12 @@ module.exports = function (grunt) {
 				args.push('-c', options.ini);
 			}
 
+			if (options.settings) {
+				for (var key in options.settings) {
+					args.splice(0, 0, '-d', key + '=' + options.settings[key]);
+				}
+			}
```
Comment (sindresorhus) [inline tasks/php.js]: You can use [`dargs`](https://github.com/sindresorhus/dargs) for this.

File: readme.md [hunk 2]
```diff
@@ -197,6 +214,13 @@ Default: `false`
 
 Suppress output produced by the PHP-server.
 
+### settings
```
Comment (sindresorhus) [inline readme.md]: Not a big fan of the option name. Maybe `directives`, `iniDirectives`, `phpDirectives`, ... ?

Comment (gglnx) [inline readme.md]: You're right. Sould be concurrent to the PHP naming. Renamed it to `directives`.

File: tasks/php.js (hunk 1 above)
Comment (gglnx) [inline tasks/php.js]: If I see it correctly, `dargs` prefixes every argument with `--`, but the CLI wants, that every directive starts with a `-d`, e.g. `-d date.timezone=Europe/Berlin -d log_errors=1 -d error_log=/var/log/php.log`

File: tasks/php.js [hunk 3]
```diff
@@ -51,7 +51,8 @@ module.exports = function (grunt) {
 			open: false,
 			bin: 'php',
 			silent: false,
-			env: {}
+			env: {},
+			settings: {}
```
Comment (sindresorhus) [inline tasks/php.js]: `directives`

File: tasks/php.js [hunk 4]
```diff
@@ -72,6 +73,12 @@ module.exports = function (grunt) {
 				args.push('-c', options.ini);
 			}
 
+			if (options.directives) {
+				for (var key in options.directives) {
+					args.splice(0, 0, '-d', key + '=' + options.directives[key]);
```
Comment (sindresorhus) [inline tasks/php.js]: Why are you splicing? Just `args.push()` the directives.

File: tasks/php.js (hunk 3 above)
Comment (gglnx) [inline tasks/php.js]: Ah. Thanks!

File: tasks/php.js (hunk 4 above)
Comment (gglnx) [inline tasks/php.js]: Japp. Changed it.

**Pull request closed without merging**